src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.cs

```diff
@@ -125,8 +125,34 @@ protected override void SetupNewTextView(IVsTextView textView)
             var openDocument = wpfTextView.TextBuffer.AsTextContainer().GetRelatedDocuments().FirstOrDefault();
             var isOpenMetadataAsSource = openDocument != null && openDocument.Project.Solution.Workspace.Kind == WorkspaceKind.MetadataAsSource;
 
+            ConditionallyCollapseOutliningRegions(textView, wpfTextView, optionsService, isOpenMetadataAsSource);
+
+            // If this is a metadata-to-source view, we want to consider the file read-only
+            IVsTextLines vsTextLines;
+            if (isOpenMetadataAsSource && ErrorHandler.Succeeded(textView.GetBuffer(out vsTextLines)))
+            {
+                ((IVsTextBuffer)vsTextLines).SetStateFlags((uint)BUFFERSTATEFLAGS.BSF_USER_READONLY);
+
+                var runningDocumentTable = (IVsRunningDocumentTable)SystemServiceProvider.GetService(typeof(SVsRunningDocumentTable));
+                var runningDocumentTable4 = (IVsRunningDocumentTable4)runningDocumentTable;
+
+                if (runningDocumentTable4.IsMonikerValid(openDocument.FilePath))
+                {
+                    var cookie = runningDocumentTable4.GetDocumentCookie(openDocument.FilePath);
+                    runningDocumentTable.ModifyDocumentFlags(cookie, (uint)_VSRDTFLAGS.RDT_DontAddToMRU | (uint)_VSRDTFLAGS.RDT_CantSave, fSet: 1);
+                }
+            }
+        }
+
+        private void ConditionallyCollapseOutliningRegions(IVsTextView textView, IWpfTextView wpfTextView, IOptionService optionsService, bool isOpenMetadataAsSource)
+        {
             var outliningManagerService = this.Package.ComponentModel.GetService<IOutliningManagerService>();
             var outliningManager = outliningManagerService.GetOutliningManager(wpfTextView);
+            if (outliningManager == null)
```

Comment:
If you didn't already, can you do a quick find refeernces on GetOutliningManager and see if we have any other offenders? This is news to me that it can be null. :-)